### DISCUSSION TITLE
Flet & Multiprocessing / Process Pool Execute

### DISCUSSION THREAD

**Discussion by ap4499** [flet-dev/flet#4162] [Q&A]
### Question

My program utilises Multiprocessing (or Process Pool Executor, which is obviously related), under some situations the GUI will duplicate, depending on how it was "built" (pack versus Build)

To run the task I use the following:

GUI thread:
- ft.page.run_task(function containing multiprocess)
- flet.app(target=main) is correctly placed with if name = main in main.py, and no other scripts have code that is unguarded by a if name block.

The thread that is spun off from ft.page.run_task():
- Multiprocess function; I have trialled both Process Pool and mutiprocess with the same results

Within the if name = main block, I use the following, as is standard when using PyInstaller/other packagers when packing for OS's that use Spawn rather than Fork for this scenario, I believe both Windows + MacOS

```
if __name__ == "__main__":
    multiprocessing.freeze_support()
```

Scenarios:
Flet Pack (Windows/Mac):
- The application runs as expected.

Flet Build (MacOS) - V0.23.2:
- The application runs as expected.

Flet Build Windows - V0.24.1/V0.25(dev):
- When running the part of the application that triggers the multi-process code, the GUI duplicates several times.

Flet Run - development environment:
- No issues.

When creating using Flet Build, is there anything that I need to do to get this working correctly?

I have been careful not to update the GUI from the subprocesses, and use a Queue to communicate status, and to debug, I have even removed all status communication entirely - with the issue still occurring. To me, the behaviour is almost as though somewhere in the code (during Flet Build), I am missing the crucial multiprocessing.freeze_support() line of code.

Has anyone come across anything similar, or is there anything I should be adding to get this working?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by necromancerWang** [comment]
same issue

**Reply by ap4499** [reply]
I raised an issue - but still no luck.

#4283 


Another thread which might be of interest:
#4252

**Comment by tynguyen2k1** [comment]
same
